lib/backports/3.2.0/enumerator/product.rb

```diff
@@ -47,15 +54,17 @@ def __execute(block, values, enums)
     def size
       total_size = 1
       @__enums.each do |enum|
+        return nil if !enum.respond_to?(:size)
         size = enum.size
-        return size if size == nil || size == Float::INFINITY
+        return size if size == nil || size == Float::INFINITY || size == -Float::INFINITY
```

Comment:
A negative size makes no sense at all, -infinity even less so. I dislike explicitly returning a clearly incorrect value, so I would prefer either apply "garbage in garbage out" (i.e. ignore negative sizes as currently), or return `nil` (but why limit this to -infinity and not to all negative numbers?).
If there is a rubyspec to that effect, I would challenge as flat out wrong. This behaviour is a bug, not a feature.

```suggestion
        return size if size == nil || size == Float::INFINITY
```

Reply:
I see that this is what Ruby does, but I don't see a spec for it (nor should there be one)

Reply:
There was a spec I fixed this for:

https://github.com/jruby/jruby/blob/730780c69f916c4fab46392a9de6124bd8b6c4b9/spec/ruby/core/enumerator/product/size_spec.rb#L26-L32

I have no stake in this so I'll leave it up to you to argue your case :-)

Reply:
FWIW, specs are often written to cover many weird and useless edge cases, frequently cases that were not even considered during the design of a feature. I am definitely form-fitting here without debating correctness.

Reply:
Oh, thanks, I didn't realize there was a separate spec for `size`. 

I opened https://github.com/ruby/spec/pull/1140